src/pydata_sphinx_theme/toctree.py

```diff
@@ -244,12 +248,7 @@ def _generate_header_nav_before_dropdown(
 
         # Wrap the final few header items in a "more" dropdown
         links_dropdown = [
-            # 🐲 brittle code because it relies on the code above to build the HTML in a particular way
-            html.replace("nav-link", "nav-link dropdown-item").replace(
-                # Prevents the header-link mixin from applying to links within the dropdown
-                "pst-header-nav-item",
-                "",
-            )
+            html.replace(nav_item, "").replace(nav_link, "nav-link dropdown-item")
             for html in links_html[n_links_before_dropdown:]
         ]
```

Comment:
No need to do it in this PR, but after merge, maybe open an issue, we likely don't need to replace now that `boilerplate` have arguments. We can  just slice `links_html` into solo and dropdown link and format differently.